Python/pywarpx/picmi.py

```diff
@@ -2028,6 +2065,28 @@ def initialize_inputs(self):
             fields_to_plot.sort()
             self.diagnostic.fields_to_plot = fields_to_plot
 
+        particle_fields_to_plot = list()
+        if self.particle_fields_to_plot is not None:
+            for pfd in self.particle_fields_to_plot:
+                if pfd.name in particle_fields_to_plot:
+                    raise Exception('A particle fields data name can not be repeated.')
+                particle_fields_to_plot.append(pfd.name)
+                self.diagnostic.__setattr__(
+                    f'particle_fields.{pfd.name}(x,y,z,ux,uy,uz)', pfd.func
+                )
+                self.diagnostic.__setattr__(
+                    f'particle_fields.{pfd.name}.do_average', pfd.do_average
+                )
+                self.diagnostic.__setattr__(
+                    f'particle_fields.{dataname}.filter(x,y,z,ux,uy,uz)', pfd.filter
```

Comment:
This should presumably `pfd.name` instead of `dataname`.

Reply:
Thanks for catching this!